src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -29532,5 +29548,213 @@ static void F2()
                 // }
                 Diagnostic(ErrorCode.ERR_RbraceExpected, "").WithLocation(27, 2));
         }
+
+        [Fact]
+        public void RefField_Assignment_AutoDefault_01()
+        {
+            var verifier = CompileAndVerify("""
+                using System;
+
+                try
+                {
+                    new RS();
+                }
+                catch (NullReferenceException)
+                {
+                    Console.Write(1);
+                }
+
+                ref struct RS
+                {
+                    ref int ri;
```

Comment:
Will the warning be reported even if the field is `ref readonly` (and hence the assignment is an error)? Consider testing that.

Reply:
Flow analysis behaves the same whether the referent is readonly or not. I'll add a test.